### DISCUSSION TITLE
Compiled versions of knitr and jupyter qmd files in a project

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#7482] [Q&A]
### Description

I have a mix of R qmd files (knitr) and python qmd files (jupyter). meta shortcodes are used. I would like to evaluate the meta shortcodes and create "compiled" versions of qmd. The knitr qmds will remain as qmd, but the jupyter qmd needs to be converted to .ipynb.

---

Here is a dummy project:

```
.
├── compiled
├── index.qmd
├── knitr.qmd
├── _metadata.yml
├── python.qmd
└── _quarto.yml
```

**_quarto.yml**

```
project:
  type: website
  output-dir: docs

metadata-files:
  - _metadata.yml
```

**_metadata.yml**

```
knitr_title: Knitr
jupyter_title: Jupyter
block: This is a block of test.
```

**knitr.qmd**

````
---
title: "{{ meta knitr_title }}"
engine: knitr
---

## Title

{{< meta block >}}

```{r}
date()
```
````

**jupyter.qmd**

````
---
title: "{{ meta jupyter_title }}"
engine: jupyter
---

## Title

{{< meta block >}}

```{python}
from datetime import date
print("Today's date:", date.today())
```
````

I would like the compiled files in the "compiled" folder.

---

Here is what I have tried;

For knitr.qmd, something like this

`quarto render knitr.qmd --to markdown-header_attributes --metadata engine:markdown`

creates the compiled qmd into the docs folder. The `--output` argument doesn't help to change the output path.

For python.qmd, I need to run the same thing

`quarto render python.qmd --to markdown-header_attributes --metadata engine:markdown`

and then

`quarto convert docs/python.md --output compiled/python.ipynb`

The `--output` actually works in this case.

This seems quite chaotic. What is the best way to do this?

Additionally, the compiled output yaml contains a long list of all meta variables. Is it possible to keep only the variables in the original qmd file? In this example, only `title` and `engine`.

```
> quarto --version
1.3.433
```

**Comment by mcanouil** [comment]
I believe this to be related to your previous question: https://github.com/quarto-dev/quarto-cli/discussions/7227

Something I do not understand, you set the output directory to "docs" but want the outputs to be in "compiled". Why not set the correct output?

See:
- https://github.com/quarto-dev/quarto-cli/discussions/2171

As it seems to be part of a bigger project (website), why not leveraging the pre-render script to render and move files before rendering the project itself?

**Reply by royfrancis** [reply]
Yes, it's related my old question. So, I want the rendered HTML to be in **docs** folder and that will be the website. The "compiled" qmds and ipynb are to be shared with students.

**Reply by mcanouil** [reply]
It's confusing, why not using code-tools or even linked Jupyter notebooks?
There are various ways:
- use pre-render to create the "compiled" documents and link those documents inside your website
- use the link format and code tools features to link Jupyter notebook and source code.

There are all described in the documentation.

**Reply by royfrancis** [reply]
code-tools looked promising, but the shortcodes are not replaced. I will look into the other options.
